docs/_articles/en/v3.x/salesforce-graph-engine/rules.md

```diff
@@ -5,21 +5,59 @@ lang: en
 
 Salesforce Graph Engine includes path-based and data-flow analysis rules.
 
-| Rule | Type | Status | Description |
-| -------- | ----------- | ----------- | ----------- |
-| ApexFlsViolationRule | Path-based analysis | Generally Available (GA) | Detects Create, Read, Update, and Delete (CRUD) and Field-Level Security violations. |
-| ApexNullPointerExceptionRule | Path-based analysis | GA | Identifies Apex operations that dereference null objects and throw NullPointerExceptions. |
-| MultipleMassSchemaLookupRule | Path-based analysis | Pilot | Detects scenarios where expensive schema lookups are made more than one time in a paths. |
-| UnimplementedTypeRule | Graph-based analysis | GA | Detects abstract classes and interfaces that are non-global and missing implementations or extensions. |
-| UnusedMethodRule | Path-based analysis | Pilot | Detects methods contained in your code that aren’t invoked from any entry points that Graph Engine recognizes. |
-
-## ApexFlsViolationRule
-ApexFlsViolationRule detects [Create, Read, Update, and Delete (CRUD) and Field-Level Security (FLS) violations](https://www.youtube.com/watch?v=1ZYjpjPTIn8). To run the path-based analysis, rules run `scanner:run:dfa`. Alternatively, run `scanner:run:dfa --category “Security”` to run only the ApexFlsViolationRule. 
-
-Example: 
+| Rule | Type | Catgegory | Status | Description |
+| -------- | ----------- | ----------- |----------- | ----------- |
+| ApexFlsViolationRule | Path-based analysis | Security | Generally Available (GA) | Detects Create, Read, Update, and Delete (CRUD) and Field-Level Security violations. |
+| ApexNullPointerExceptionRule | Path-based analysis | Error-Prone | GA | Identifies Apex operations that dereference null objects and throw NullPointerExceptions. |
+| AvoidDatabaseOperationInLoop | Path-based analysis | Performance | Pilot | Detects database operations in loops that degrade performance. |
+| AvoidMultipleMassSchemaLookups | Path-based analysis | Performance | GA | Detects scenarios where expensive schema lookups are made more than one time in a paths. |
+| RemoveUnusedMethod | Path-based analysis | Performance | Pilot | Detects methods contained in your code that aren’t invoked from any entry points that Graph Engine recognizes. |
+| UnimplementedTypeRule | Graph-based analysis | Performance | GA | Detects abstract classes and interfaces that are non-global and missing implementations or extensions. |
+| UseWithSharingOnDatabaseOperation | Path-based analysis | Security | Pilot | Detects database operations outside with-sharing-annotated classes. |
+
+## Running Graph Engine GA Rules <a name='RunningGraphEngineGARules'>#</a>
+Run all Graph Engine rules against your code, or run a subset of rules by type or by category.
+
+To run the path-based rules run: `scanner:run:dfa --projectdir MyDirectory`.
+
+**Example:**
+  
+```sfdx scanner:run:dfa --projectdir /project/dir --target /project/dir/target1```
+<br>
```

Comment:
The `<br>`s at this line, line 32, and line 38 need to have an empty line above them, for whatever reason. Otherwise they still render bunched together weirdly.

Reply:
done